pkg/operation/etcdencryption/encryptionconfiguration.go

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package encryptionconfiguration
+
+import (
+	"fmt"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	apiserverconfigv1 "k8s.io/apiserver/pkg/apis/config/v1"
+)
+
+// NewPassiveConfiguration creates a initial configuration for etcd encryption
+// The list of encryption providers contains identity as first provider, which has
+// the effect, that this configuration does not yet encrypt written secrets. The
+// configuration has to be activated to actually encrypt written secrets.
+// Nevertheless, an encryption provider aescbc is already contained in the configuration
+// at the second position in the list of providers. A key is created for aescbc with
+// the key's name containing the timestamp when it was created in UTC.
+//
+// apiVersion: apiserver.config.k8s.io/v1
+// kind: EncryptionConfiguration
+// resources:
+// - providers:
+//   - aescbc:
+//       keys:
+//       - name: key1559747207815249000
+//         secret: Y8LEzbtK/2mdXrw6W/faAxNLu+mTCmcQeWojShAJGEg=
+//   - identity: {}
+//   resources:
+//     metadata:
+//   - secrets
+func NewPassiveConfiguration() (*apiserverconfigv1.EncryptionConfiguration, error) {
+	key, err := NewEncryptionKey()
+	if err != nil {
+		return nil, err
+	}
+	return &apiserverconfigv1.EncryptionConfiguration{
+		TypeMeta: v1.TypeMeta{
```

Comment:
Do we need this? Usually, if you're using the `versioning.NewDefaultingCodec` and a proper serializer, you shouldn't need this.

Reply:
In my small test coding
````go
func TestAbs(t *testing.T) {
	ec := &apiserverconfigv1.EncryptionConfiguration{}
	t.Log(ec.Kind)
	t.Log(ec.APIVersion)
}
````
the kind and apiversion are not filled ...

Reply:
Yes they are not set but will be if you use a proper serializer / codec.